src/modm/processing/fiber/barrier.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2024, Niklas Hauser
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include "fiber.hpp"
+
+namespace modm::fiber
+{
+
+/// @ingroup modm_processing_fiber
+/// @{
+
+/// Implements the `std::barrier` interface for fibers.
+/// @warning This implementation is not interrupt-safe!
+/// @see https://en.cppreference.com/w/cpp/thread/barrier
+template< class CompletionFunction = decltype([]{}) >
+class barrier
+{
+	barrier(const barrier&) = delete;
+	barrier& operator=(const barrier&) = delete;
+	using count_t = uint16_t;
+
+	const CompletionFunction completion;
+	count_t expected;
+	count_t count;
+	volatile count_t sequence{};
+
+public:
+	using arrival_token = count_t;
+
+	constexpr explicit
+	barrier(std::ptrdiff_t expected, CompletionFunction f = CompletionFunction())
+	: completion(std::move(f)), expected(expected), count(expected) {}
+
+	[[nodiscard]]
+	static constexpr std::ptrdiff_t
+	max() { return count_t(-1); }
```

Comment:
```suggestion
	max() { return std::numeric_limits<count_t>::max(); }
```
A bit cleaner than the cast to get the maximum.